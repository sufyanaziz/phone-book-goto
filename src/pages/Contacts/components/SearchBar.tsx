import React, { useContext } from "react";
import { Theme, css } from "@emotion/react";
import { ContactStore } from "@common/store/useContactStore";
import Button from "@common/components/Button";
import { useNavigate } from "react-router-dom";

const SearchBar = () => {
  const navigate = useNavigate();
  const searchBarStyle = (theme: Theme) => {
    return css({
      width: "100%",
      marginBottom: "16px",
      display: "flex",
      justifyContent: "space-between",
      ".input": {
        padding: theme.padding[1],
        width: "200px",
      },
    });
  };

  const { onSetSearch, search } = useContext(ContactStore);

  return (
    <div css={searchBarStyle}>
      <input
        className="input"
        onChange={(e) => onSetSearch(e.target.value)}
        value={search}
        placeholder="Find Contact"
      />
      <div>
        <Button label="Add New Contact" onClick={() => navigate("/form")} />
      </div>
    </div>
  );
};

export default SearchBar;
