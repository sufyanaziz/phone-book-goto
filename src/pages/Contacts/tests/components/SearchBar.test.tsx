import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "@pages/Contacts/components/SearchBar";

describe("Component SearchBar", () => {
  it("should render SearchBar component correctly", () => {
    const component = renderer.create(<SearchBar />);
    expect(component).toMatchSnapshot();
  });
});
