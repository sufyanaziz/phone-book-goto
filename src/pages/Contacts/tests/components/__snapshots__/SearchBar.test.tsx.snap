// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchBar should render SearchBar component correctly 1`] = `
<div
  css={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Find Contact"
    value=""
  />
  <div>
    <button
      className="primary"
      css={[Function]}
      onClick={[Function]}
    >
      Add New Contact
    </button>
  </div>
</div>
`;
